### PR TITLE
Clarifai : Added Clarifai as LLM and embedding model provider.

### PR DESCRIPTION
**Pull request by mogith-pn**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Please delete options that are not relevant.

- [x] Unit Test
- [x] Test Script (please provide)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

File: docs/components/embedding-models.mdx [hunk 1]
```diff
@@ -66,6 +67,54 @@ embedder:
...
+
+```yaml config.yaml
+llm:
+ provider: clarifai
+ config:
+   model: "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct"
+   model_kwargs:
+     temperature: 0.5
+     max_tokens: 1000  
+
```
Comment (deshraj) [inline docs/components/embedding-models.mdx:110]:
```suggestion

```

File: docs/components/llms.mdx [hunk 2]
```diff
@@ -286,6 +287,56 @@ llm:
...
+
+```yaml config.yaml
+llm:
+ provider: clarifai
+ config:
+   model: "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct"
+   model_kwargs:
+     temperature: 0.5
+     max_tokens: 1000  
+
```
Comment (deshraj) [inline docs/components/llms.mdx:331]:
```suggestion
```

File: embedchain/embedder/clarifai.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            from clarifai.client.model import Model
+            from clarifai.client.input import Inputs
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "The required dependencies for ClarifaiEmbeddingFunction are not installed."
+                'Please install with `pip install --upgrade "embedchain[clarifai]"`'
+            ) from None
+        self.config = config
+        self.api_key = config.api_key or os.getenv("CLARIFAI_PAT")
+        self.model = config.model or "https://clarifai.com/clarifai/main/models/BAAI-bge-base-en-v15"
```
Comment (deshraj) [inline embedchain/embedder/clarifai.py:23]:
```suggestion
        self.model = config.model
```

File: embedchain/llm/clarifai.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    def _get_answer(prompt: str, config: BaseLlmConfig) -> str:
+        try:
+            from clarifai.client.model import Model
+        except ModuleNotFoundError:
+            raise ModuleNotFoundError(
+                "The required dependencies for Clarifai are not installed."
+                'Please install with `pip install --upgrade "embedchain[clarifai]"`'
+            ) from None
+
+        model_name = config.model or "https://clarifai.com/openai/chat-completion/models/GPT-4"
```
Comment (deshraj) [inline embedchain/llm/clarifai.py:32]:
```suggestion
        model_name = config.model
```

File: configs/clarifai.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  config: 
+    model: "https://clarifai.com/mistralai/completion/models/mistral-7B-Instruct"
+    model_kwargs: 
+      temperature: 0.5
+      max_tokens: 1000
+
+embedder:
+  provider: clarifai
+  config: 
+    model: "https://clarifai.com/clarifai/main/models/BAAI-bge-base-en-v15"
```
Comment (deshraj) [inline configs/clarifai.yaml:12]:
```suggestion
    model: "https://clarifai.com/clarifai/main/models/BAAI-bge-base-en-v15"

```

**Review by deshraj — APPROVED** [review]
This is great. Thanks for adding this @mogith-pn

**Comment by deshraj** [comment]
@mogith-pn can you please run `poetry lock --no-update` so that the poetry.lock file is updated? Seems like the CI is failing because of this.

**Comment by mogith-pn** [comment]
@deshraj 
Thanks for the quick review. I have executed `poetry lock --no-update` and committed the changes.
Please take a look into it.

**Comment by mogith-pn** [comment]
@deshraj ,
Addressed linting issues, could you please re-run the workflows. 
TIA :)

**Comment by mogith-pn** [comment]
@deshraj ,
I have adjusted the test code, meanwhile I couldn't able to run the tests locally, I'm getting the following error
``` RuntimeError: Session factory is not initialized. Call setup_engine() first.``` error.

**Comment by deshraj** [comment]
@mogith-pn can you try to run `rm -rf ~/.embedchain` and try running tests again? It should work then

**Comment by mogith-pn** [comment]
> @mogith-pn can you try to run `rm -rf ~/.embedchain` and try running tests again? It should work then

Thanks @deshraj !
Please take a loot at it now. I have fixed the errors.

**Comment by deshraj** [comment]
Seems like there are some lint issues. Can you please fix it by running `make lint format`

**Comment by mogith-pn** [comment]
@deshraj 
Apologies !
Addressed the lint issues.

**Comment by deshraj** [comment]
Still seeing the issue:

```
embedchain/vectordb/qdrant.py:3:8: F401 [*] `uuid` imported but unused
```

Can you please remove the UUID import? Thanks!

**Comment by mogith-pn** [comment]
@deshraj 
Is it really needed, because when i ran `poetry run ruff .` I'm getting all checks passed as result. And I also noticed there's an existing [PR](https://github.com/embedchain/embedchain/pull/1329) which addresses this ?

**Comment by deshraj** [comment]
Yeah, can you please fix it in your PR? We won't be able to merge the PR unless the CI checks pass.

**Comment by mogith-pn** [comment]
> Yeah, can you please fix it in your PR? We won't be able to merge the PR unless the CI checks pass.

Hi @deshraj ,
The issue is fixed, could you please re-run the CI.
TIA :)

**Comment by mogith-pn** [comment]
> > Yeah, can you please fix it in your PR? We won't be able to merge the PR unless the CI checks pass.
> 
> Hi @deshraj , The issue is fixed, could you please re-run the CI. TIA :)

Hey @deshraj ,
Hope you are doing well. Could you please initiate re-run for the tests. TIA :)

**Comment by mogith-pn** [comment]
Hi @deshraj / @taranjeet , 
Could you please re-initiate the workflow ? 
Thanks :)

**Comment by Dev-Khant** [comment]
Hi @mogith-pn Can you resolve the conflicts so that it can be merged?

**Comment by mogith-pn** [comment]
> Hi @mogith-pn Can you resolve the conflicts so that it can be merged?

Hey @Dev-Khant, 
could you please check this now ?

**Comment by Dev-Khant** [comment]
@mogith-pn Please fix this conflicts, once it's done we can run the tests and merge this.

**Comment by mogith-pn** [comment]
@Dev-Khant , Resolved conflicts !.

**Comment by Dev-Khant** [comment]
Hi @mogith-pn Can you please check why the tests are failing?

**Comment by mogith-pn** [comment]
> Hi @mogith-pn Can you please check why the tests are failing?

@Dev-Khant , Sorry my bad, there's inconsistency between poetry.lock and pyproject.toml !
Fixing it.

**Comment by mogith-pn** [comment]
@Dev-Khant , Kindly check now !

**Comment by Dev-Khant** [comment]
@mogith-pn Thanks a lot for quickly responding and making fixes. Sorry for the delay in running CI build tests. 
@deshraj will review this ASAP and it will be merged.

Thanks a lot for contributing!

**Comment by mogith-pn** [comment]
> @mogith-pn Thanks a lot for quickly responding and making fixes. Sorry for the delay in running CI build tests. @deshraj will review this ASAP and it will be merged.
> 
> Thanks a lot for contributing!

Sure, Happy to contribute ! Looking forward to it.

**Comment by Dev-Khant** [comment]
@mogith-pn Can you please resolve conflicts with poetry?

File: docs/components/llms.mdx [hunk 6]
```diff
@@ -12,6 +12,7 @@ Embedchain comes with built-in support for various popular large language models
   <Card title="Azure OpenAI" href="#azure-openai"></Card>
   <Card title="Anthropic" href="#anthropic"></Card>
   <Card title="Cohere" href="#cohere"></Card>
+  <Card title="Clarifai" href="#clarifai"></Card>
```
Comment (deshraj) [inline docs/components/llms.mdx:15]: Can you please move it after vLLM?

**Review by deshraj — COMMENTED** [review]
Thanks for adding this. Can you please resolve some of my comments and then we are good to go?

**Comment by mogith-pn** [comment]
@deshraj @Dev-Khant ,
Made changes ! Please take a look into it.

**Review by deshraj — APPROVED** [review]
LGTM!

**Pull request merged**